src/DOMBuilder.js

```diff
@@ -10,10 +10,11 @@
 module.exports = (function(){
   "use strict";
 
-  function DOMBuilder(document, disallowActiveAttributes) {
-    this.document = document;
-    this.fragment = document.createDocumentFragment();
-    this.currentNode = this.fragment;
+  var DocumentFragment = require("./DocumentFragment");
+
+  function DOMBuilder(disallowActiveAttributes) {
+    this.fragment = new DocumentFragment();
+    this.currentNode = this.fragment.node;
     this.contexts = [];
     this.pushContext("html", 0);
```

Comment:
In terms of slowparse, that's what we advertise, but *technically* this is not true. HTML5 is SUPER loose, and one of the things that is *not* required is an actual `<html>` element =)

See http://www.w3.org/TR/html5/semantics.html#semantics, "Tag omission in text/html". Almost all tags you'd think were necessary are actually optional, provided there are no comment tags anywhere.